declaring type in docs to match argspec

mddeff opened the pull request:
This is to fix an issue in upstream so it can be backported to 2.9 per @samccann in the [partnering merge request](https://github.com/ansible/ansible/pull/76220#issuecomment-969318633)

From my original MR:
---
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Updates docs to reflect correct `lines` argument `type` of `list` instead of default `string`.  Verified by [the module itself](https://github.com/ansible/ansible/blob/stable-2.9/lib/ansible/modules/network/edgeos/edgeos_config.py#L281)


##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Docs Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
`edgeos_config` module
##### ADDITIONAL INFORMATION
Causes issues with [SchemaStore](https://github.com/SchemaStore/schemastore/) that gets its ansible schema automatically generated from this repo. SchemaStore is used by the [RedHat VScode-yaml plug-in](https://github.com/redhat-developer/vscode-yaml) which, in turn, is used by the [Ansible VScode module](https://github.com/ansible/vscode-ansible).

First time contributor, so if I'm pointing this in the wrong direction, let me know.  Thanks!

samccann commented:
For whomever picks this up to review - Please also consider if this is the correct change in the matching PR for 2.9 by approving or commenting on https://github.com/ansible/ansible/pull/76220#issuecomment-969318633  thanks!

jplitza commented:
LGTM
Could be accompanied by `elements: str`

dericcrago commented:
hi @mddeff, thanks for this PR. What do you think about the suggestion in https://github.com/ansible-collections/community.network/pull/326#issuecomment-972890618 ?

`elements: str`

mddeff commented:
Sorry for the delay, OBE by holidays.  Added `elements: str`.  Let me know if there are any other issues. Added to [original PR](https://github.com/ansible/ansible/pull/76220) as well.

jplitza commented:
The failing CI Test 3 seems relevant:
> 01:11 ERROR: tests/sanity/ignore-2.13.txt:784:1: A100: Ignoring 'parameter-list-no-elements' on 'plugins/modules/network/edgeos/edgeos_config.py' is unnecessary (75%)

mddeff commented:
@jplitza I'm assuming the fix is removing those ignores in [tests/sanity/ignore-2.13.txt](https://github.com/ansible-collections/community.network/blob/main/tests/sanity/ignore-2.13.txt#L783-L785)?  If I'm reading it right, its specifically ignoring the validation kicking back an `type: list` with no `element` definition.  And now that it has one, its failing?

If I'm missing something please let me know, it's my first Ansible PR and I'm not as familiar with the CI setup.

jplitza commented:
> @jplitza I'm assuming the fix is removing those ignores in [tests/sanity/ignore-2.13.txt](https://github.com/ansible-collections/community.network/blob/main/tests/sanity/ignore-2.13.txt#L783-L785)? If I'm reading it right, its specifically ignoring the validation kicking back an `type: list` with no `element` definition. And now that it has one, its failing?
> 
> If I'm missing something please let me know, it's my first Ansible PR and I'm not as familiar with the CI setup.

Can't say that I'm familiar with it, but I understood it 100% the same. ;-)

samccann commented:
I agree that's what the error sounds like to me as well. Worth taking it out of the ignores and seeing what happens!

mddeff commented:
@jplitza @samccann , I may have gotten a little overly ambitious; if I remove the other ignores (besides 2.13), it looks like now we're failing on other stuff because the ignores are no longer there...

I reverted to just the changes in 2.13. Input welcome!

jplitza commented:
You removed all ignores regarding the whole module, but only the "parameter-list-no-elements" test was completely fixed by your changes.

Either you add back the "doc-missing-type" and "parameter-type-not-in-doc" ignores, or you declare types for the whole spec. ;-)

But I guess all 5 ignore files should be modified alike: Either remove the ignore from all or none.

dericcrago commented:
thanks @mddeff @jplitza @samccann !
